src/options.rs

```diff
@@ -136,25 +140,69 @@ impl Options {
     )
   }
 
-  pub(crate) fn bitcoin_rpc_client(&self) -> Result<Client> {
-    let cookie_file = self
-      .cookie_file()
-      .map_err(|err| anyhow!("failed to get cookie file path: {err}"))?;
+  fn derive_var(
+    arg_value: Option<&str>,
+    env_key: Option<&str>,
+    config_value: Option<&str>,
+    default_value: Option<&str>,
+  ) -> Result<Option<String>> {
+    let env_value = match env_key {
+      Some(env_key) => match env::var(format!("ORD_{env_key}")) {
+        Ok(env_value) => Some(env_value),
+        Err(err @ env::VarError::NotUnicode(_)) => return Err(err.into()),
+        Err(env::VarError::NotPresent) => None,
+      },
+      None => None,
+    };
+
+    Ok(
+      arg_value
+        .or(env_value.as_deref())
+        .or(config_value)
+        .or(default_value)
+        .map(str::to_string),
+    )
+  }
+
+  pub(crate) fn auth(&self) -> Result<Auth> {
+    let config = self.load_config()?;
+
+    let rpc_user = Options::derive_var(
+      self.bitcoin_rpc_user.as_deref(),
+      Some("BITCOIN_RPC_USER"),
+      config.rpc_user.as_deref(),
+      None,
+    )?;
+
+    let rpc_pass = Options::derive_var(
+      self.bitcoin_rpc_pass.as_deref(),
+      Some("BITCOIN_RPC_PASS"),
+      config.rpc_pass.as_deref(),
+      None,
+    )?;
+
+    match (rpc_user, rpc_pass) {
+      (Some(rpc_user), Some(rpc_pass)) => Ok(Auth::UserPass(rpc_user, rpc_pass)),
```

Comment:
We should handle the `None/Some` and `Some/None` cases by throwing an error, and we should have tests for both of these cases.